orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/expressions/PipelineExpressionEvaluatorSpec.groovy

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline.expressions
+
+import com.netflix.spinnaker.kork.expressions.ExpressionFunctionProvider
+import com.netflix.spinnaker.orca.pipeline.model.Execution
+import spock.lang.Specification
+
+class PipelineExpressionEvaluatorSpec extends Specification {
+
+  def 'should set execution aware functions for the given function providers'() {
+
+    given: 'function providers'
+    ExpressionFunctionProvider expressionFunctionProvider1 = buildExpressionFunctionProvider('pipeline')
+    ExpressionFunctionProvider expressionFunctionProvider2 = buildExpressionFunctionProvider('jenkins')
+
+    when: 'registered with pipeline evaluator'
+    PipelineExpressionEvaluator evaluator = new PipelineExpressionEvaluator([expressionFunctionProvider1, expressionFunctionProvider2])
+
+    then:
+    noExceptionThrown()
+    evaluator
```

Comment:
what does this line do?

Reply:
Just checks the object to be a non null. Probably redundant with that noExceptionThrown check.. Will remove it.

Reply:
ah, didn't know that syntax, cool! but yeah, prob not needed esp since you deref it right after